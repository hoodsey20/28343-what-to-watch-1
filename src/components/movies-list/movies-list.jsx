import React from 'react';
import PropTypes from 'prop-types';

import withVideo from '../../hocs/with-video/with-video';
import SmallMoovieCard from '../small-movie-card/small-movie-card.jsx';

const SmallMoovieCardWithVideo = withVideo(SmallMoovieCard);

const MoviesList = ({
  movies,
  cardHoverHandler,
  cardLeaveHandler,
  activeCard,
}) => {
  return (
    <div className="catalog__movies-list">
      {movies.map((it) => <SmallMoovieCardWithVideo
        key={it.id}
        isPlaying={activeCard === it.id}
        movie={it}
        hoverHandler={cardHoverHandler}
        leaveHandler={cardLeaveHandler}
        clickHandler={(movie) => {
          /*eslint-disable*/
          console.log(`фильм: `, movie)
          /*eslint-disable*/
        }}
      />)}
    </div>
  );
}

export default MoviesList;

MoviesList.propTypes = {
  movies: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
        cover: PropTypes.string.isRequired,
      })
  ).isRequired,
  activeCard: PropTypes.number,
  cardHoverHandler: PropTypes.func.isRequired,
  cardLeaveHandler: PropTypes.func.isRequired,
};
