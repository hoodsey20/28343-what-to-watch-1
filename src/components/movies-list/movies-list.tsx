import * as React from "react";
import * as PropTypes from "prop-types";
import { Movie } from "../../types";

import SmallMoovieCard from "../small-movie-card/small-movie-card";
import withActivePlayer from "../../hocs/with-active-player/with-active-player";
import withShowMore from "../../hocs/with-show-more/with-show-more";

interface Props {
  movies: Movie[];
  history: any;
  cardHoverHandler: () => void;
  cardLeaveHandler: () => void;
  activeCard: Movie["id"];
}

export const MoviesList: React.SFC<Props> = ({
  movies,
  cardHoverHandler,
  cardLeaveHandler,
  activeCard,
  history
}) => {
  return (
    <div className="catalog__movies-list">
      {movies.map(it => (
        <SmallMoovieCard
          key={it.id}
          isPlaying={activeCard === it.id}
          movie={it}
          hoverHandler={cardHoverHandler}
          leaveHandler={cardLeaveHandler}
          clickHandler={movie => {
            history.push(`/film/${movie.id}`);
          }}
        />
      ))}
    </div>
  );
};

export default withActivePlayer(withShowMore(MoviesList));
