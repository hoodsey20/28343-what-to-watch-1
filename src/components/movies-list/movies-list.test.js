import React from 'react';
import renderer from 'react-test-renderer';

import {MoviesList} from './movies-list.jsx';

const mock = [
  {
    id: 1,
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    cover: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    src: [`https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
      `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`]
  },
  {
    id: 2,
    name: `Bohemian Rhapsody`,
    cover: `img/bohemian-rhapsody.jpg`,
    src: [`https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
      `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`]
  },
];

it(`MoviesList is rendered correctly`, () => {
  const tree = renderer.create(<MoviesList
    movies={mock}
    onClick={jest.fn()}
    cardHoverHandler={jest.fn()}
    cardLeaveHandler={jest.fn()}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
