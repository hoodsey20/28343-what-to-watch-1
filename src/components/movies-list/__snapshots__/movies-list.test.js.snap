// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList is rendered correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[MockFunction]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="250"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[MockFunction]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster="img/bohemian-rhapsody.jpg"
        width="250"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
</div>
`;
